ArdeleanuStefan_Task3/src/test.cpp

```diff
@@ -0,0 +1,67 @@
+#define BOOST_TEST_MODULE TEST_CLIENT_SERVER_IPC
+#define BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
+#include <fstream>
+#include <iostream>
+std::ifstream client1{"tmp/log.0"};
+std::ifstream client2{"tmp/log.1"};
+
+BOOST_AUTO_TEST_SUITE(ClientServerIPCTest)
+
+BOOST_AUTO_TEST_CASE(test_1)
+{
+    system("./src/script.sh ");
+    system("echo 'add;unu\nget;0\ndelete;doi' > tmp/input.pipe");
+    system("echo 'add;doi\nget;1\ndelete;doi' > tmp/input1.pipe");
+
+    std::string output;
+    int count = 0;
+    while (std::getline(client1, output))
+    {
+        switch (count)
+        {
+        case 0:
+            BOOST_CHECK(output == "Add OK");
```

Comment:
If the client1 file is empty or has only one of the expected checks the test will pass.